halo2-wasm/src/metadata.rs

```diff
@@ -0,0 +1,177 @@
+    pub is_aggregation: bool,
+
+    // RlcCircuitParams:
+    /// The number of advice columns per phase
+    pub num_advice_per_phase: Vec<u16>,
+    /// The number of special advice columns that have range lookup enabled per phase
+    pub num_lookup_advice_per_phase: Vec<u8>,
+    /// Number of advice columns for the RLC custom gate
+    pub num_rlc_columns: u16,
+    /// The number of fixed columns **only** for constants
+    pub num_fixed: u8,
+
+    // This is specific to the current Verify Compute Circuit implementation and provided just for data availability:
+    /// The maximum number of user outputs. Used to determine where to split off `compute_snark`'s public instances between user outputs and data subqueries.
+    /// This does **not** include the old accumulator elliptic curve points if
+    /// the `compute_snark` is from an aggregation circuit.
+    pub max_outputs: u16,
+}
+
+impl AxiomV2CircuitMetadata {
+    pub fn encode(&self) -> anyhow::Result<H256> {
+        let mut encoded = vec![];
+        encoded.write_u8(self.version)?;
+
+        encoded.write_u8(self.num_instance.len().try_into()?)?;
+        for &num_instance in &self.num_instance {
+            encoded.write_u32::<BigEndian>(num_instance)?;
+        }
+
+        let num_phase = self.num_challenge.len();
+        if num_phase == 0 {
+            bail!("num_challenge must be non-empty")
+        }
+        encoded.write_u8(num_phase.try_into()?)?;
+        for &num_challenge in &self.num_challenge {
+            encoded.write_u8(num_challenge)?;
+        }
+
+        encoded.write_u8(self.is_aggregation as u8)?;
+
+        // encode RlcCircuitParams:
+        if self.num_advice_per_phase.len() > num_phase {
+            bail!("num_advice_per_phase must be <= num_phase")
+        }
+        let mut num_advice_cols = self.num_advice_per_phase.clone();
+        num_advice_cols.resize(num_phase, 0);
+        for num_advice_col in num_advice_cols {
+            encoded.write_u16::<BigEndian>(num_advice_col)?;
+        }
+
+        if self.num_lookup_advice_per_phase.len() > num_phase {
+            bail!("num_lookup_advice_per_phase must be <= num_phase")
+        }
+        let mut num_lookup_advice_cols = self.num_lookup_advice_per_phase.clone();
+        num_lookup_advice_cols.resize(num_phase, 0);
+        for num_lookup_advice_col in num_lookup_advice_cols {
+            encoded.write_u8(num_lookup_advice_col)?;
+        }
+
+        encoded.write_u16::<BigEndian>(self.num_rlc_columns)?;
+        encoded.write_u8(self.num_fixed)?;
+
+        encoded.write_u16::<BigEndian>(self.max_outputs)?;
+
+        if encoded.len() > 32 {
+            bail!("circuit metadata cannot be packed into bytes32")
+        }
+        encoded.resize(32, 0);
+        Ok(H256::from_slice(&encoded))
+    }
+
+    pub fn from_circuit_params(
```

Comment:
nit: name this `from_base_circuit_params` and comment that we hardcoded `num_challenge: vec![0]`